src/cpp/src/continuous_batching_pipeline.cpp

```diff
@@ -60,6 +60,15 @@ class ContinuousBatchingPipeline::Impl {
     ChatHistory m_history;
 
 
+    void _notify_requests_dropped_by_handle() {
+        // Notify the last time by pushing empty output
+        // This causes read() to unblock by adding anything to the queue
```

Comment:
Even with this comment I don't fully understand why do we need to send empty outputs? 
If handle is dropped by user, user should not expect any outputs from this request / handle

Reply:
It's additional protection in multithreading scenarios. When generation handle is dropped from thread #1 and thread #2 is blocked on read() this operation unlocks it. That's the case in model server where handle drop is called from a callback triggered by HTTP server on client disconnect.